albumentations/augmentations/functional.py

```diff
@@ -2041,3 +2041,44 @@ def glass_blur(img, sigma, max_delta, iterations, dxy, mode):
             x[h, w], x[h + dy, w + dx] = x[h + dy, w + dx], x[h, w]
 
     return np.clip(cv2.GaussianBlur(x / coef, sigmaX=sigma, ksize=(0, 0)), 0, 1) * coef
+
+
+def fourier_domain_adaptation(img, target_img, beta):
+    """
+    Fourier Domain Adaptation from https://github.com/YanchaoYang/FDA
+
+    Args:
+        img:  source image
+        target_img:  target image for domain adaptation
+        beta: coefficient from source paper
+
+    Returns:
+        transformed image
+
+    """
+    # get fft of both source and target
+    fft_src = np.fft.fft2(img.astype(np.float32), axes=(0, 1))
+    fft_trg = np.fft.fft2(target_img.astype(np.float32), axes=(0, 1))
+
+    # extract amplitude and phase of both fft-s
+    amplitude_src, phase_src = np.abs(fft_src), np.angle(fft_src)
+    amplitude_trg = np.abs(fft_trg)
+
+    # mutate the amplitude part of source with target
+    amplitude_src = np.fft.fftshift(amplitude_src, axes=(0, 1))
+    amplitude_trg = np.fft.fftshift(amplitude_trg, axes=(0, 1))
+    height, width = amplitude_src.shape[:2]
+    border = np.floor(min(height, width) * beta).astype(int)
+    center_x, center_y = np.floor([height / 2.0, width / 2.0]).astype(int)
+
+    y1, y2 = center_y - border, center_y + border + 1
+    x1, x2 = center_x - border, center_x + border + 1
+
+    amplitude_src[y1:y2, x1:x2] = amplitude_trg[y1:y2, x1:x2]
+    amplitude_src = np.fft.ifftshift(amplitude_src, axes=(0, 1))
+
+    # get mutated image
+    src_image_transformed = np.fft.ifft2(amplitude_src * np.exp(1j * phase_src), axes=(0, 1))
+    src_image_transformed = np.real(src_image_transformed)
+
+    return np.clip(src_image_transformed, 0, 255).astype(np.uint8)
```

Comment:
If store image datatype at function start and change this line this transform will supports any datatype.
```python
return np.clip(src_image_transformed, 0, 255).astype(dtype)
```

Reply:
Do not forget after that change docstring about image types

Reply:
done